### PR TITLE
Adding Non-DBGap project usersync Tests

### PR DESCRIPTION
**Pull request by atharvar28**
https://ctds-planx.atlassian.net/browse/PXP-9688

### CODE REVIEW

File: services/apis/nondbgapUsersync/nonDbgapTasks.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    I.click({ xpath: 'xpath: //button[contains(text(), \'Sign in\')]' });
+    I.saveScreenshot('AfterSignIn.png');
+    // checking if the authcode is present in the url after login
+    I.seeInCurrentURL('code');
+    // grabbing the authcode from the url and storing it in the variable
+    const authCodeURL = await I.grabCurrentUrl();
+    const url = new URL(authCodeURL);
+    const authCode = url.searchParams.get('code');
+    // check if the authCode isnt empty
+    expect(authCode).to.not.to.be.empty;
```
Comment (jingh8) [inline services/apis/nondbgapUsersync/nonDbgapTasks.js:53]: use chai.expect or add this `const { expect } = chai;` at start?

File: services/apis/nondbgapUsersync/nonDbgapTasks.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      id = indexdRecordDID.data.records[0].id;
+    }
+    // return the id
+    return id;
+  },
+
+  async presignedURLRequest(fence, project, projectFile, user, token) {
+    console.log(`Creating presigned url with mainAcct user for ${project} after running usersync`);
+    const signedURLs3 = await fence.do.createSignedUrl(projectFile, ['protocol=s3'], token);
+    if (signedURLs3.data.status === 200) {
```
Comment (jingh8) [inline services/apis/nondbgapUsersync/nonDbgapTasks.js:110]: To my knowledge, to get status code from a response should use `signedURLs3.status`. response.data will return the response body.

**Review by jingh8 — APPROVED** [review]

File: services/apis/nondbgapUsersync/nonDbgapTasks.js (hunk 1 above)
Comment (atharvar28) [reply services/apis/nondbgapUsersync/nonDbgapTasks.js:53]: Yes used the wrong import 
https://github.com/uc-cdis/gen3-qa/pull/818/files#diff-2dac60cb7af5a9a9349ef245b6bfda9231fb4b8c22f160fa66b4deecc66a07d7R1

File: hooks/test_conditions.js [hunk 3]
```diff
@@ -47,5 +47,18 @@ module.exports = async function () {
         });
       }
     }
+
+    if (suite.title === 'nonDBGap Project Usersync @requires-fence @requires-indexd') {
+      const dbGaPConfig = bash.runCommand('gen3 secrets decode fence-config fence-config.yaml | yq -r \'.dbGaP[] | ."info".study_to_resource_namespaces | ."PROJECT-12345"\'');
```
Comment (atharvar28) [inline hooks/test_conditions.js:52]: still unsure if this is the right way of checking when to run the test in test conditions. Still have to talk to Binam about what can be best way of checking

**Review by jingh8 — DISMISSED** [review]

File: suites/apis/nonDbgapUsersyncTest.js [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+      port: 22
+      proxy: 'cloud-proxy.internal.io'
+      proxy_user: 'sftpuser'
+      #decrypt_key: 'NCBI-DBGAP-TrustedPartner'
+      encrypted: false
+      study_to_resource_namespaces:
+        '_default': ['/']
+        'PROJECT-12345': ['']
+        'PROJECT-67890': ['']
+    allow_non_dbgap_whitelist: true
```
Comment (BinamB) [inline suites/apis/nonDbgapUsersyncTest.js:26]:
```suggestion
    allow_non_dbGaP_whitelist: true
```

**Review by jingh8 — APPROVED** [review]

**Review by haraprasadj — APPROVED** [review]
+1

**Pull request merged**